### PR TITLE
kotlin cleanup

### PR DESCRIPTION
**Pull request by criticalAY**
## Pull Request template
Kotlin Cleanup 

## Purpose / Description
Renaming variable and using scope function as kotlin cleanup

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt [hunk 1]
```diff
@@ -76,29 +76,33 @@ class DeckOptionsActivity :
...
-                mValues["name"] = deck.getString("name")
-                mValues["desc"] = deck.getString("desc")
-                mValues["deckConf"] = deck.getString("conf")
-                // general
-                mValues["maxAnswerTime"] = mOptions.getString("maxTaken")
-                mValues["showAnswerTimer"] = parseTimerValue(mOptions).toString()
-                mValues["autoPlayAudio"] = mOptions.getBoolean("autoplay").toString()
-                mValues["replayQuestion"] = mOptions.optBoolean("replayq", true).toString()
+                mValues.apply {
+                    "name" to deck.getString("name")
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt:85]: This code is not equivalent with the original. `"name" to deck.getString("name")` simply creates a Pair<String, String> and it doesn't update the mValues map. You can see this as you can do : 

`val unusedByMap = "name" to deck.getString("name")`

Use the set() method as that is the one that enables the previous bracket notation:

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt [hunk 2]
```diff
@@ -76,29 +76,33 @@ class DeckOptionsActivity :
...
                 val newOptions = mOptions.getJSONObject("new")
-                mValues["newSteps"] = StepsPreference.convertFromJSON(newOptions.getJSONArray("delays"))
-                mValues["newGradIvl"] = newOptions.getJSONArray("ints").getString(0)
-                mValues["newEasy"] = newOptions.getJSONArray("ints").getString(1)
-                mValues["newFactor"] = (newOptions.getInt("initialFactor") / 10).toString()
-                mValues["newOrder"] = newOptions.getString("order")
-                mValues["newPerDay"] = newOptions.getString("perDay")
-                mValues["newBury"] = newOptions.optBoolean("bury", true).toString()
+                newOptions.apply {
+                    "newSteps" to StepsPreference.convertFromJSON(newOptions.getJSONArray("delays"))
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt:98]: Same as above with set(), this time on a JSONObject.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt [hunk 3]
```diff
@@ -115,25 +119,31 @@ class DeckOptionsActivity :
...
                 val lapOptions = mOptions.getJSONObject("lapse")
-                mValues["lapSteps"] = StepsPreference.convertFromJSON(lapOptions.getJSONArray("delays"))
-                mValues["lapNewIvl"] = String.format(Locale.ROOT, "%.0f", lapOptions.getDouble("mult") * 100)
-                mValues["lapMinIvl"] = lapOptions.getString("minInt")
-                mValues["lapLeechThres"] = lapOptions.getString("leechFails")
-                mValues["lapLeechAct"] = lapOptions.getString("leechAction")
-                // options group management
-                mValues["currentConf"] = col.decks.getConf(deck.getLong("conf"))!!.getString("name")
+                lapOptions.apply {
+                    "lapSteps" to StepsPreference.convertFromJSON(lapOptions.getJSONArray("delays"))
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt:123]: Same as above

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt [hunk 4]
```diff
@@ -115,25 +119,31 @@ class DeckOptionsActivity :
...
                 if (mOptions.has("reminder")) {
                     val reminder = mOptions.getJSONObject("reminder")
                     val reminderTime = reminder.getJSONArray("time")
 
-                    mValues["reminderEnabled"] = reminder.getBoolean("enabled").toString()
-                    mValues["reminderTime"] = String.format(
-                        "%1$02d:%2$02d", reminderTime.getLong(0), reminderTime.getLong(1)
-                    )
+                    mValues.apply {
+                        "reminderEnabled" to reminder.getBoolean("enabled").toString()
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt:137]: I would revert this change and the one immediate bellow this.
While apply {} looks nice I think it's overkill when you have only two calls on the same object. You're basically doubling the previous number of lines of code with questionable increase in readability.

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivity.kt [hunk 5]
```diff
@@ -324,11 +324,12 @@ class MultimediaEditFieldActivity : AnkiActivity(), OnRequestPermissionsResultCa
         imageFieldController!!.showCropDialog(content) { saveAndExit() }
     }
 
-    @KotlinCleanup("scope function")
     private fun saveAndExit(ignoreField: Boolean = false) {
         val resultData = Intent()
-        resultData.putExtra(EXTRA_RESULT_FIELD, if (ignoreField) null else mField)
-        resultData.putExtra(EXTRA_RESULT_FIELD_INDEX, mFieldIndex)
+        resultData.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivity.kt:329]: `val resultData = Intent(). apply {}`

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt [hunk 6]
```diff
@@ -297,8 +307,10 @@ class NoteServiceTest : RobolectricTest() {
 
         // case: all cards are new or learning
         for (card in note.cards()) {
-            card.type = newOrLearningList.shuffled().first()
-            card.flush()
+            card.apply {
+                type = newOrLearningList.shuffled().first()
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:311]: See above comment on using apply with only two calls on the same variable.

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt [hunk 7]
```diff
@@ -262,8 +268,10 @@ class NoteServiceTest : RobolectricTest() {
 
         // test case: all cards are new
         for (card in note.cards()) {
-            card.type = Consts.CARD_TYPE_NEW
-            card.flush()
+            card.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:271]: See above comment on using apply with only two calls on the same variable.

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt [hunk 8]
```diff
@@ -231,8 +233,10 @@ class NoteServiceTest : RobolectricTest() {
         val file = createTransientFile("foo")
 
         val field = ImageField()
-        field.extraImagePathRef = file.absolutePath
-        field.hasTemporaryMedia = true
+        field.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:236]: `val field = ImageField.apply {}`

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt [hunk 9]
```diff
@@ -217,8 +217,10 @@ class NoteServiceTest : RobolectricTest() {
         val file = createTransientFile("foo")
 
         val field = MediaClipField()
-        field.audioPath = file.absolutePath
-        field.hasTemporaryMedia = true
+        field.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:220]: `val field = MediaClipField().apply {}`

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt [hunk 10]
```diff
@@ -112,7 +112,7 @@ class NoteServiceTest : RobolectricTest() {
     @Test
     @Throws(IOException::class)
     fun importImageToDirectoryTest() {
-        val fileImage = directory.newFile("testimage.png")
+        val fileImage = directory.newFile("test_image.png")
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:115]: Revert this change and keep the original name.

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt [hunk 11]
```diff
@@ -68,8 +66,10 @@ class NoteServiceTest : RobolectricTest() {
         multiMediaNote.getField(1)!!.text = "bar"
 
         val basicNote = Note(testCol!!, testModel)
-        basicNote.setField(0, "this should be changed to foo")
-        basicNote.setField(1, "this should be changed to bar")
+        basicNote.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:69]: `val basicNote = Note(testCol!!, testModel).apply {}`

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt (hunk 10 above)
Comment (criticalAY) [reply AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:115]: There was a annotation to rename it?

Comment (criticalAY) [reply AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:114]: and I think it is correct as per naming convention no?

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt [hunk 12]
```diff
@@ -76,52 +77,59 @@ class DeckOptionsActivity :
...
-                mValues["name"] = deck.getString("name")
-                mValues["desc"] = deck.getString("desc")
-                mValues["deckConf"] = deck.getString("conf")
-                // general
-                mValues["maxAnswerTime"] = mOptions.getString("maxTaken")
-                mValues["showAnswerTimer"] = parseTimerValue(mOptions).toString()
-                mValues["autoPlayAudio"] = mOptions.getBoolean("autoplay").toString()
-                mValues["replayQuestion"] = mOptions.optBoolean("replayq", true).toString()
+                mValues.apply {
+                    set("name", "name")
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt:86]: Why "name" for the second parameter when originally it was deck.getString("desc")?
The same for the following two calls.

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt (hunk 10 above)
Comment (lukstbit) [reply AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:114]: Leave it as it is. With "annotation to rename it" I think you're referring to the type raised by the IDE, but that can be ignored.

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt (hunk 10 above)
Comment (criticalAY) [reply AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:114]: There was a kotlin cleanup annotation mentioning the renaming

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt (hunk 12 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckOptionsActivity.kt:86]: mistakenly did that

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt (hunk 10 above)
Comment (lukstbit) [reply AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.kt:114]: @criticalAY You're right, I completely missed the class annotation. The file should be renamed.

**Review by lukstbit — APPROVED** [review]
Thanks!

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt [hunk 13]
```diff
@@ -380,9 +379,11 @@ open class Anki2Importer(col: Collection?, file: String) : Importer(col!!, file)
...
                 // copy it over
                 val model = srcModel.deepClone()
-                model.put("id", mid)
-                model.put("mod", TimeManager.time.intTime())
-                model.put("usn", mCol.usn())
+                model.apply {
+                    put("id", mid)
+                    put("mod", TimeManager.time.intTime())
+                    put("usn", mCol.usn())
+                }
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt:386]:
```kotlin
val model = srcModel.deepClone().apply {
...
}
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt [hunk 14]
```diff
@@ -392,9 +393,11 @@ open class Anki2Importer(col: Collection?, file: String) : Importer(col!!, file)
...
                 // they do; we can reuse this mid
                 val model = srcModel.deepClone()
-                model.put("id", mid)
-                model.put("mod", TimeManager.time.intTime())
-                model.put("usn", mCol.usn())
+                model.apply {
+                    put("id", mid)
+                    put("mod", TimeManager.time.intTime())
+                    put("usn", mCol.usn())
+                }
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt:400]: ditto

File: AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt [hunk 15]
```diff
@@ -519,12 +518,13 @@ open class RobolectricTest : CollectionGetter {
...
     protected fun openDialogFragmentUsingActivity(menu: DialogFragment): FragmentTestActivity {
         val startActivityIntent = Intent(targetContext, FragmentTestActivity::class.java)
-        val activity = startActivityNormallyOpenCollectionWithIntent(FragmentTestActivity::class.java, startActivityIntent)
-        activity.showDialogFragment(menu)
-        return activity
+        return startActivityNormallyOpenCollectionWithIntent(
+            FragmentTestActivity::class.java, startActivityIntent
+        ).apply {
+            showDialogFragment(menu)
+        }
```
Comment (BrayanDSO) [inline AnkiDroid/src/test/java/com/ichi2/anki/RobolectricTest.kt:527]: Personally, I find the previous code easier to read. No need for the change

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

**Review by BrayanDSO — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**